scheduler/scheduler.go

```diff
@@ -61,6 +70,61 @@ func NewScheduler(tasks []Task, s storage.StorageApi, e event.EventApi, p pwd.PW
 	return sch, nil
 }
 
+func (s *scheduler) updatePlaygrounds() {
+	s.mx.Lock()
+	defer s.mx.Unlock()
+
+	log.Printf("Updating playgrounds configuration\n")
+	for playgroundId, _ := range s.playgrounds {
+		playground, err := s.storage.PlaygroundGet(playgroundId)
+		if err != nil {
+			log.Printf("Could not find playground %s\n", playgroundId)
+			continue
+		}
+		s.playgrounds[playgroundId] = playground
+		matchedTasks := s.getMatchedTasks(playground)
+		s.playgroundTasks[playground.Id] = matchedTasks
+	}
+}
+
+func (s *scheduler) schedulePlaygroundsUpdate() {
+	s.updatePlaygrounds()
+	s.ticker = time.NewTicker(time.Minute * 5)
+	go func() {
+		for range s.ticker.C {
+			s.updatePlaygrounds()
+		}
+	}()
+}
+
+func (s *scheduler) getMatchedTasks(playground *types.Playground) []Task {
+	matchedTasks := []Task{}
+	for _, expr := range playground.Tasks {
+		for _, task := range s.tasks {
+			if expr == task.Name() {
+				matchedTasks = append(matchedTasks, task)
+				continue
+			}
+			matched, err := regexp.MatchString(expr, task.Name())
+			if err != nil {
+				continue
+			}
+			if matched {
+				matchedTasks = append(matchedTasks, task)
+				continue
+			}
+		}
+	}
+	return matchedTasks
+}
+
+func (s *scheduler) getTasks(playgroundId string) []Task {
+	s.mx.Lock()
```

Comment:
We're blocking everything (all the playgrounds) just to retrieve the tasks for only one playground. I'm aware that this should be extremely fast.. but still IDK if it's the best approach.

Reply:
You are more than welcome to propose a better approach :) As long as it is simple.